torch/distributed/algorithms/model_averaging/averagers.py

```diff
@@ -4,7 +4,29 @@
 import torch.distributed.algorithms.model_averaging.utils as utils
 
 
-class PeriodicModelAverager:
+class ModelAverager:
+    r"""Base class for all model averagers.
+
+    Args:
+        params (Iterator[torch.nn.Parameter]): The model parameters to be averaged.
+        process_group: The process group to be used for all-reduce.
+                       If ``None``, the default process group, which
+                       is created by :func:`torch.distributed.init_process_group`,
+                       will be used. (default: ``None``)
+    """
+
+    def __init__(self, params, process_group=None):
+        self.params = list(params)
+        self.process_group = (
+            process_group if process_group is not None else dist.group.WORLD
+        )
+        self.step = 0
+
+    def average_parameters(self):
```

Comment:
Should we consider making `ModelAverager` an abstract class and enforcing that `average_parameters` is overridden, or at least raise a NotImplementedError here?